src/search.rs

```diff
@@ -0,0 +1,141 @@
+use crate::document::advanced::DocumentEncryptUnmanagedResult;
+use crate::document::advanced::*;
+use crate::document::DocumentEncryptOpts;
+use crate::internal::{take_lock, IronOxideErr};
+use crate::{GroupId, IronOxide};
+use async_trait::async_trait;
+use rand::{self, RngCore};
+use std::convert::{TryFrom, TryInto};
+use std::ops::DerefMut;
+
+use ironcore_search_helpers::generate_hashes_for_string;
+
+type Result<A> = std::result::Result<A, IronOxideErr>;
+///The required length of the salt.
+const REQUIRED_LEN: usize = 32;
+
+///The result of creating a new index as well as indexing on a particular phrase.
+///If you only want to create the index, see create_index.
+pub struct CreatedIndexResult {
+    pub encrypted_salt: DocumentEncryptUnmanagedResult,
+    pub sdk: IronSimpleSearch,
+}
+
+///Trait which gives the ability to create an index
+#[async_trait]
+pub trait SimpleSeachInitialize {
+    ///Given the encrypted salt and the edeks, decrypt them and give back the IronSimpleSearch object.
+    async fn initialize_search_index(
+        &self,
+        encrypted_data: &[u8],
+        encrypted_deks: &[u8],
+    ) -> Result<IronSimpleSearch>;
+    ///Create an index and encrypt it to the provided group_id.
+    ///If you need to index terms immediately, see `create_index_and_initialize` which will return
+    ///the IronSimpleSearch for reuse.
+    async fn create_index(&self, group_id: &GroupId) -> Result<DocumentEncryptUnmanagedResult>;
```

Comment:
Shouldn't the `DocumentEncryptUnmanagedResult` that this returns be the input `initialize_search_index`?

Reply:
The `DocumentEncryptUnmanagedResult` has the people it was encrypted to, errors about the sharing, etc. Maybe I should change the structure to just include the relevant data.

Reply:
Or maybe give it an alias that better describes its usage here?